manager/main.go

```diff
@@ -62,34 +57,62 @@ func init() {
 	//+kubebuilder:scaffold:scheme
 }
 
-func initFlags(fs *pflag.FlagSet) {
-	fs.StringVar(&metricsAddr, "metrics-bind-address", "localhost:8080", "The address the metric endpoint binds to.")
-	fs.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
-	fs.BoolVar(&enableLeaderElection, "leader-elect", false,
+type config struct {
+	metricsAddr          string
+	enableLeaderElection bool
+	probeAddr            string
+	gates                []string
+	logging              *logsv1.LoggingConfiguration
+}
+
+func newConfig() *config {
+	c := &config{
+		logging: logsv1.NewLoggingConfiguration(),
+	}
+	c.logging.Format = logsv1.JSONLogFormat
+	c.logging.Verbosity = logsv1.VerbosityLevel(4)
+
+	return c
+}
+
+func initFlags(fs *pflag.FlagSet, config *config) {
+	logsv1.AddFlags(config.logging, fs)
+
+	fs.StringVar(&config.metricsAddr, "metrics-bind-address", "localhost:8080", "The address the metric endpoint binds to.")
+	fs.StringVar(&config.probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	fs.BoolVar(&config.enableLeaderElection, "leader-elect", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
-	fs.StringSliceVar(&gates, "feature-gates", []string{}, "A set of key=value pairs that describe feature gates for alpha/experimental features. ")
+	fs.StringSliceVar(&config.gates, "feature-gates", []string{}, "A set of key=value pairs that describe feature gates for alpha/experimental features. ")
 }
 
 func main() {
-	opts := kzap.Options{
-		Development: true,
-		Level:       zap.NewAtomicLevelAt(-4),
-	}
-	opts.BindFlags(flag.CommandLine)
+	config := newConfig()
 	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
-
-	initFlags(pflag.CommandLine)
+	initFlags(pflag.CommandLine, config)
 	pflag.Parse()
-	ctrl.SetLogger(kzap.New(kzap.UseFlagOptions(&opts)))
-	features.FeedGates(gates)
+
+	// Temporary logger for initialization
+	setupLog := ctrl.Log.WithName("setup")
+
+	if err := logsv1.ValidateAndApply(config.logging, nil); err != nil {
```

Comment:
Is this how the logging gets assigned to all the reconcilers?

Reply:
This reads the logging config (that can be overridden with manager flags) and inits (if necessary) a logger configured accordingly.

The reconciler don't get a logger assigned. `controller-runtime` injects a logger in the context on each reconciliation loop.